backstory/src/test/java/com/google/sps/story/NLServiceClientTest.java

```diff
@@ -0,0 +1,387 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.story;
+
+import static org.mockito.Mockito.*;
+
+import com.google.cloud.language.v1.AnalyzeSyntaxResponse;
+import com.google.cloud.language.v1.Document;
+import com.google.cloud.language.v1.LanguageServiceClient;
+import com.google.cloud.language.v1.PartOfSpeech;
+import com.google.cloud.language.v1.PartOfSpeech.Form;
+import com.google.cloud.language.v1.PartOfSpeech.Proper;
+import com.google.cloud.language.v1.PartOfSpeech.Tag;
+import com.google.cloud.language.v1.Token;
+import com.google.sps.story.data.NLServiceClient;
+import com.google.sps.story.data.WordType;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.ArgumentCaptor;
+import org.mockito.ArgumentMatchers;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+/**
+ * Quality tests for the NLServiceClient
+ */
+@RunWith(JUnit4.class)
+public final class NLServiceClientTest {
+
+  /** array of multiword inputs to use for testing */
+  final String[] MULTIWORD_INPUTS = {"physical exercise", "social group",
+        "urban area", "human leg", "tower block", "metropolitan area"};
+  /** array of nouns to use for testing */
+  private final String[] NOUNS = { "person", "dog", "skyscraper" };
+  /** array of proper nouns to use for testing */
+  private final String[] PROPER_NOUNS = { "Mary", "Michael", "AnneMarie", "Maddie" };
+  /** array of adjectives to use for testing */
+  private final String[] ADJECTIVES = { "calm", "fun", "happy", "blue" };
+  /** gerunds to use for testing */
+  private final String[] GERUNDS = { "baking", "jogging", "running"};  
+
+  /**
+   * Check that an exception thrown if you try to create
+   * a NLServiceClient with null LanguageServiceClient.
+   */
+  @Test (expected = IllegalArgumentException.class)
+  public void nullClient() {
+    NLServiceClient client = new NLServiceClient(null);
+  }
+
+  /**
+   * Check that document passed in to analyze has
+   * the correct text.
+   */
+  @Test 
+  public void checkDocumentText() {
+    final String TEXT = "calm";
+    PartOfSpeech partOfSpeech = PartOfSpeech.newBuilder().setTag(Tag.ADJ).build();
+
+    Map<String, PartOfSpeech> desiredReturns = new HashMap<String, PartOfSpeech>();
+    desiredReturns.put(TEXT, partOfSpeech);
+    
+    LanguageServiceClient mockClient = createMockLSClient(desiredReturns);
+
+    NLServiceClient client = new NLServiceClient(mockClient);
+    client.sortByWordType(Arrays.asList(TEXT));
```

Comment:
Is there a way to verify that two `analyzeSyntax()` calls are made if this argument has two items?

Reply:
Yes, there is. I just added two test methods (`checkAPICalls` and `checkAPICallsWithGerund`) to do just that.

Reply:
Done